connectivity/suite.go

```diff
@@ -352,6 +398,34 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
 			return check.ResultDrop, check.ResultNone
 		})
 
+	// Test L7 HTTP named port introspection using an egress policy on the clients.
+	ct.NewTest("client-egress-l7-named-port").
+		WithFeatureRequirements(check.RequireFeatureEnabled(check.FeatureL7Proxy)).
+		WithPolicy(clientEgressOnlyDNSPolicyYAML).         // DNS resolution only
+		WithPolicy(clientEgressL7HTTPNamedPortPolicyYAML). // L7 allow policy with HTTP introspection (named port)
+		WithScenarios(
+			tests.PodToPod(),
+			tests.PodToWorld(),
+		).
+		WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+			if a.Source().HasLabel("other", "client") && // Only client2 is allowed to make HTTP calls.
+				// Outbound HTTP to one.one.one.one is L7-introspected and allowed.
+				(a.Destination().Port() == 80 && a.Destination().Address() == "one.one.one.one" ||
+					a.Destination().Port() == 8080) { // 8080 is traffic to echo Pod.
+				if a.Destination().Path() == "/" || a.Destination().Path() == "" {
```

Comment:
would add to the end of the comment: `via named port`. Otherwise this validation code is kind of hard to read w.r.t. where a named port was used.